executor/executor.go

```diff
@@ -1042,12 +1043,20 @@ func (e *SelectLockExec) Next(ctx context.Context, req *chunk.Chunk) error {
 	for id := range e.tblID2Handle {
 		e.updateDeltaForTableID(id)
 	}
-
-	return doLockKeys(ctx, e.ctx, newLockCtx(e.ctx.GetSessionVars(), lockWaitTime, len(e.keys)), e.keys...)
+	lockCtx, err := newLockCtx(e.ctx, lockWaitTime, len(e.keys))
+	if err != nil {
+		return err
+	}
+	return doLockKeys(ctx, e.ctx, lockCtx, e.keys...)
 }
 
-func newLockCtx(seVars *variable.SessionVars, lockWaitTime int64, numKeys int) *tikvstore.LockCtx {
-	lockCtx := tikvstore.NewLockCtx(seVars.TxnCtx.GetForUpdateTS(), lockWaitTime, seVars.StmtCtx.GetLockWaitStartTime())
+func newLockCtx(sctx sessionctx.Context, lockWaitTime int64, numKeys int) (*tikvstore.LockCtx, error) {
+	seVars := sctx.GetSessionVars()
+	forUpdateTS, err := sessiontxn.GetTxnManager(sctx).GetStmtForUpdateTS()
+	if err != nil {
+		return nil, err
+	}
+	lockCtx := tikvstore.NewLockCtx(forUpdateTS, lockWaitTime, seVars.StmtCtx.GetLockWaitStartTime())
```

Comment:
You changed the old code process of `newLockCtx`. The first parameter of `NewLockCtx` comes from "seVars.TxnCtx.GetForUpdateTS()" before, but now from "sessiontxn.GetTxnManager(sctx).GetStmtForUpdateTS()". 
Does the old code has some bugs?

Reply:
The change it not due to a bug fix. This PR aims to change the management of the acquisition of ts by txnManager. Previsouly, some ts acquisition if from txnManager, but some is from TxnCtx such as here.

Reply:
But the logic of "seVars.TxnCtx.GetForUpdateTS()"  is different from "sessiontxn.GetTxnManager(sctx).GetStmtForUpdateTS()".

Reply:
Yes, previsouly, executor can change TxnCtx.forUpdateTS directly. Now, forUpdateTS is managed by the relevant transaction context provider and returned by GetStmtForUpdateTS. Although when forUpdateTS is changed, we also change txnCtx.forUpdateTS, now we should use GetTxnManager(sctx).GetStmtForUpdateTS() in normal code and only use TxnCtx.GetForUpdateTS() in places such as log print.

Reply:
> GetTxnManager(sctx).GetStmtForUpdateTS() in normal code and only use TxnCtx.GetForUpdateTS() in places such as log print.

Any plan to remove the usage of `GetForUpdateTS()` in point_get and batch_point_get?

Reply:
Some will be removed in the later PRs.